bd-logger/tests/logger_integration.rs

```diff
@@ -1323,6 +1341,79 @@ mod tests {
     );
   }
 
+  #[test]
+  fn transforms_emitted_logs_according_to_filters() {
+    let mut setup = Setup::new();
+
+    setup.send_runtime_update(true, true, false);
+
+    // Send down a configuration:
+    //  * with a single buffer ('default') which accepts all logs
+    //  * a single workflow which flushes all buffers when it sees a log with field "foo" equal to
+    //    'fire workflow action!'
+    //  * a filter that adds a field "foo" with value 'fire workflow action!'
+    let maybe_nack = setup.send_configuration_update(config_helper::configuration_update(
```

Comment:
might be nice to have this take a struct instead of a bunch of Option args, then you can use ..Default() to only set a few fields